client/src/components/CreateTrip.tsx

```diff
@@ -68,3 +98,22 @@ export default function CreateTrip(): JSX.Element {
     </>
   );
 }
+
+type InputProps = {
+  name: string;
+  id: string;
+  label: string;
+  type?: string;
+};
+
+function TextInput({ label, ...props }: InputProps): JSX.Element {
+  const [field, meta] = useField(props);
```

Comment:
dint understand this bit..

Reply:
Thats a hook that formik provides, to connect the form level data with the input field. The field is a set of props related to things like the input value, the onChange handler etc.
The meta is a set of props for meta data about the field, like whether it's currently got a valid entry or not, whether the user has touched it, etc.